### PR TITLE
[#153300848] hide orders/transactions/trade button if the broker doesn't support it

### PR DESCRIPTION
**Pull request by guillaumedebavelaere**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift [hunk 1]
```diff
@@ -24,6 +26,7 @@ class TradeItPortfolioAccountDetailsViewController: TradeItViewController, Trade
         self.tableViewManager.delegate = self
         self.tableViewManager.table = self.table
 
+        self.fetchBrokerSupportedServices()
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift]: What if the user changes account to select a different broker?

File: TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift [hunk 2]
```diff
@@ -191,3 +211,25 @@ class TradeItPortfolioAccountDetailsViewController: TradeItViewController, Trade
...
+    }
+
+    func supportsService(broker: TradeItBroker) -> Bool {
+        switch self {
+        case .orders: return broker.supportsOrderStatus()
+        case .trading: return broker.supportsTrading()
+        case .transactions: return broker.supportsTransactionsHistory()
+        }
+    }
+}
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift]: Nice!

File: TradeItIosTicketSDK2/TradeItTradingConfirmationViewController.swift [hunk 3]
```diff
@@ -25,7 +25,20 @@ import UIKit
...
+                        return broker.shortName == self.order?.linkedBrokerAccount?.brokerName
+                    }
+                ) else {
+                    return
+                }
+                self.viewOrderStatusButton.isHidden = (!TradeItSDK.isPortfolioEnabled || !broker.supportsOrderStatus())
+            }, onFailure: { _ in
+                self.viewOrderStatusButton.isHidden = true
+            }
+        )
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTradingConfirmationViewController.swift]: We seem to be checking capabilities a few times in the SDK now - do you think it's a good time to pull this broker search in to a new function yet?

File: TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift (hunk 1 above)
Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift]: We hit again `viewDidLoad` and this method

File: TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift [hunk 4]
```diff
@@ -147,22 +147,38 @@ class TradeItPortfolioAccountDetailsViewController: TradeItViewController, Trade
...
+        }
+        TradeItSDK.linkedBrokerManager.getBroker(
+            shortName: brokerShortName,
+            onSuccess: { broker in
+                self.brokerSupportedService = [
+                    (supportedService: SupportedService.orders, handler: self.orderActionWasTapped),
+                    (supportedService: SupportedService.transactions, handler: self.transactionActionWasTapped),
+                    (supportedService: SupportedService.trading, handler: self.tradeActionWasTapped)
+                ].filter {$0.supportedService.supportsService(broker: broker) }
+                self.brokerSupportedService.count > 0 ? (self.activityButton.isEnabled = true) : (self.activityButton.isEnabled = false)
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItPortfolioAccountDetailsViewController.swift]: `self.activityButton.isEnabled = self.brokerSupportedService.count > 0`

**Pull request merged**